RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.cc

```diff
@@ -118,16 +116,16 @@ std::vector<CSCSegment> CSCSegAlgoRU::buildSegments(const ChamberHitContainer& u
     }
 
     //change the tresholds if it's time to look for displaced mu segments
-    if(doCollisions && search_disp && int(rechits.size()-used_rh)>2){//check if there are enough recHits left to build a segment from displaced vertices
-      doCollisions = false;
+    if(l_doCollisions && search_disp && int(rechits.size()-used_rh)>2){//check if there are enough recHits left to build a segment from displaced vertices
```

Comment:
Andrea, good eyes. I think this is strictly a bug as you say. But used_rh should be converted to unsigned before the subtraction and I think that will be fine: rechits.size() is all the rechits available to process so it is always larger than the number 'used' in a segment.

I'm still of the opinion that non-reproducibility is most likely from the initialization manipulations, and Nikolay is working on clearing that up.